```{r}
library(dplyr)
library(DESeq2)
library(readr)
library(dplyr)
library(ggplot2)
```

```{r}
# Reading in the data with specified column types --------------------------------------------
# Specify column types for gene counts
mycounts <- read_csv("C:/Users/Sheryar Malik/Downloads/RNa-seq analysis/Anotherdataset/gene_counts_cleaned.csv", 
                     col_types = cols(
  gene = col_character(),
  UM_N01 = col_double(),
  UM_P01 = col_double(),
  UM_P02 = col_double(),
  UM_N02 = col_double(),
  UM_P03 = col_double(),
  UM_P04 = col_double(),
  UM_P05 = col_double(),
  UM_P06 = col_double(),
  UM_N03 = col_double(),
  UM_N04 = col_double(),
  UM_N05 = col_double(),
  UM_N06 = col_double(),
  UM_P07 = col_double(),
  UM_N07 = col_double(),
  UM_N08 = col_double(),
  UM_P08 = col_double(),
  UM_N09 = col_double(),
  UM_N10 = col_double(),
  UM_P09 = col_double(),
  UM_P10 = col_double(),
  UM_P11 = col_double(),
  UM_N11 = col_double(),
  UM_P12 = col_double(),
  UM_N12 = col_double(),
  UM_N13 = col_double(),
  UM_P13 = col_double(),
  UM_N14 = col_double(),
  UM_P14 = col_double(),
  UM_P15 = col_double(),
  UM_P16 = col_double(),
  UM_P17 = col_double(),
  UM_N15 = col_double(),
  UM_N16 = col_double(),
  UM_N17 = col_double(),
  UM_N18 = col_double(),
  UM_P18 = col_double()
))

# Specify column types for metadata
metadata <- read_csv("C:/Users/Sheryar Malik/Downloads/RNa-seq analysis/Anotherdataset/metadata.csv", 
                     col_types = cols(
                       sample_id = col_character(),
                       group = col_character()
                     ))

# Convert the data to a dataframe 
mycounts <- as.data.frame(mycounts)
metadata <- as.data.frame(metadata)

# Display the first few rows of gene counts to verify
head(mycounts, 32)
```
```{r}
# Convert the data to a dataframe 
mycounts <- as.data.frame(mycounts)
metadata <- as.data.frame(metadata)
```

```{r}
# Display the first few rows of gene counts to verify
head(mycounts, 32)
```

```{r}
# remove duplicate data if any
library(readr)

data <- read_csv("C:/Users/Sheryar Malik/Downloads/RNa-seq analysis/Anotherdataset/gene_counts.csv")  # Load your data

duplicates <- data[duplicated(data$gene), ]  # Replace 'your_row_name_column' with the actual column name you think might have duplicates
print(duplicates)

```

```{r}
library(readr)
library(dplyr)

# Load your data
data <- read_csv("C:/Users/Sheryar Malik/Downloads/RNa-seq analysis/Anotherdataset/gene_counts.csv")

# Remove the rows with 'gene' equal to '1-Mar' or '2-Mar'
data_cleaned <- data %>%
  filter(!(gene %in% c("1-Mar", "2-Mar")))

# Save the cleaned data to a new CSV file
write_csv(data_cleaned, "C:/Users/Sheryar Malik/Downloads/RNa-seq analysis/Anotherdataset/gene_counts_cleaned.csv")

# Print a message to confirm that the data has been saved
cat("Cleaned data saved to gene_counts_cleaned.csv\n")


```


```{r}
# Prepare DESeq2 dataset -- replace 'Type' with your actual column name for conditions
dds <- DESeqDataSetFromMatrix(countData = mycounts, 
                              colData = metadata, 
                              design = ~ group,
                              tidy = TRUE)
```



```{r}
# Assuming `dds` is your DESeqDataSet object
dds <- estimateSizeFactors(dds)

# Get normalized counts
normlzd_dds <- counts(dds, normalized = TRUE)

# Boxplot of normalized data
boxplot(normlzd_dds, main = "Expression after normalization")

# Save normalized counts to a new CSV file
write.csv(normlzd_dds, "C:/Users/Sheryar Malik/Downloads/RNa-seq analysis/Anotherdataset/normalized_counts.csv")

# Print a message to confirm that the normalized data has been saved
cat("Normalized counts saved to normalized_counts.csv\n")
```

```{r}
# DESeq2 Analysis ------------------------------------------------
dds <- DESeq(dds)
resultsNames(dds)  # Check contrast names
```



```{r}
# Set comparison groups - replace "HC" and "Tumor" with control and case labels #in my case i am using normal and disease tag in heaty control sectoin
#only change conditon = group and all relvnt data
res <- results(dds, contrast = c("group", "hc", "disease"), tidy = TRUE)
res <- res[order(res$padj), ]  # Order by adjusted p-value
res <- res %>% mutate(sig = padj < 0.05)
```


```{r}
# DATA EXPLORATION ----------------------------------------------
# PCA Plot
vsdata <- vst(dds, blind = FALSE)
pca_data <- plotPCA(vsdata, intgroup = c("group"), returnData = TRUE)
plotPCA(vsdata, intgroup = c("group"))
ggplot(pca_data, aes(x = PC1, y = PC2, color = group, label = name)) +
  geom_point() +
  geom_text()
```


```{r}
# Dispersion and MA plot
plotDispEsts(dds, main = "Dispersion Estimates")
plotMA(dds)
```
```{r}
# VOLCANO PLOT ---------------------------------------------------
res %>%
  filter(!is.na(padj)) %>%
  ggplot(aes(x = log2FoldChange, y = -log10(padj), color = padj < 0.05 & abs(log2FoldChange) > 1)) +
  scale_colour_manual(values = c("gray", "red")) +
  geom_point(size = 0.5) +
  geom_hline(yintercept = -log10(0.05)) +
  geom_vline(xintercept = 1) +
  geom_vline(xintercept = -1)
```

```{r}
# PLOT COUNTS FOR SPECIFIC GENES ---------------------------------
# Plot individual gene expression (replace with your genes of interest)
plotCounts(dds, gene = "TGFBI", intgroup = "group")
TGFBI <- plotCounts(dds, gene = "TGFBI", intgroup = "group", returnData = TRUE)
gn.most.diff.val <- counts(dds, normalized = TRUE)["TGFBI",]
barplot(gn.most.diff.val, col = "black", main = "TGFBI", las = 2, cex.names = 0.5, cex.lab = 2)
```
```{r}
# Export DEGs ---------------------------------------------------
# Export significant DEGs with adjusted p-value < 0.05
res %>%
  filter(padj < 0.05) %>%
  write_csv("C:/Users/Sheryar Malik/Downloads/RNa-seq analysis/Anotherdataset/DEGs.csv")

res
dim(res)
```









































































































